source/Irrlicht/CSkinnedMesh.cpp

```diff
@@ -813,6 +813,21 @@ bool CSkinnedMesh::setHardwareSkinning(bool on)
 	return HardwareSkinning;
 }
 
+void CSkinnedMesh::refreshJointCache()
+{
+	//copy cache from the mesh...
+	for (u32 i=0; i<AllJoints.size(); ++i)
+	{
+		SJoint *joint=AllJoints[i];
+		for (u32 j=0; j<joint->Weights.size(); ++j)
+		{
+			const u16 buffer_id=joint->Weights[j].buffer_id;
+			const u32 vertex_id=joint->Weights[j].vertex_id;
+			joint->Weights[j].StaticPos = LocalBuffers[buffer_id]->getVertex(vertex_id)->Pos;
+			joint->Weights[j].StaticNormal = LocalBuffers[buffer_id]->getVertex(vertex_id)->Normal;
+		}
+	}
+}

```

Comment:
Might need a call to `normalizeWeights`? Not sure.

Reply:
normalizeWeights depends on joint strength, but not vertex normals or vertex positions. Looks like it's necessary as part of finalize(), but not here.